### DISCUSSION TITLE
Problem with importing Clash configurations in the "Husi" application.

### DISCUSSION THREAD

**Discussion by ghost** [xchacha20-poly1305/husi#26] [General]
Hello.  There is a problem in the operation of your application "husi" - the text Clash configurations of the clipboard and the file itself with the extension ".yml" are not imported.  If this can be fixed, please fix it.  The application is very good.

**Comment by xchacha20-poly1305** [comment]
为什么有的 _非 Clash 软件_ 会选择支持 Clash 的配置格式？原因之一是历史遗留问题，因为许多 **机场黑产** 甚至只提供 Clash 配置文件，并且由于 **非标准** 分享链接的滥用以及部分分享链接标准模糊，导入分享链接通常不能获得较为完整的配置信息，此时导入 Clash 配置就能获得较为完整的配置信息。

适配机场是个一举两得的壮举：第一可以满足大量只想使用一个机场订阅器的 **最终用户** 的需求，第二是可以吸引部分 「合作」 机场。但是，有多少机场是通过\*ray 或 sing-box 来过墙的？恐怕寥寥无几。助长对这些机场的需求以及帮助这些机场本身只会削弱开源社区的热情。

现在时代变了，sing-box 官方客户端也是 **仅** 支持 sing-box 配置，一些第三方 GUI 也是仅支持 sing-box 订阅（如 [GUI.for.SingBox](https://github.com/GUI-for-Cores/GUI.for.SingBox)）。仅支持导入 sing-box 的配置，不仅能获得准确的配置信息，还能促进原生支持 sing-box 的相关社区发展，更能对抱守旧制的机场黑产体制发起新的挑战。

既然如此，为何不果断抛弃历史包袱？如果您对 sing-box 不信任，那么请您不要使用基于 sing-box 的服务。

**Reply by ghost** [reply]
Your application is visually very similar in interface and functions to such well-known V2Ray clients as "NekoBox" and "Matsuri", and they do not have such a problem with importing yaml files and data.  But your answer is clear to me.  Thank you for your reply.🙏
